src/backend/src/routes/projects.routes.ts

```diff
@@ -47,6 +48,31 @@ projectRouter.post(
 projectRouter.post('/:wbsNum/set-team', nonEmptyString(body('teamId')), validateInputs, ProjectsController.setProjectTeam);
 projectRouter.delete('/:wbsNum/delete', ProjectsController.deleteProject);
 projectRouter.post('/:wbsNum/favorite', ProjectsController.toggleFavorite);
+projectRouter.post(
+  '/material/:wbsNum/create',
+  nonEmptyString(body('name')),
+  nonEmptyString(body('assemblyId').optional()),
+  nonEmptyString(
```

Comment:
you can make a custom validation method for this, see how we do isWorkPackageStageOrNone